File: src/main/java/com/ahold/technl/sandbox/entity/Delivery.java
package com.ahold.technl.sandbox.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

import java.util.UUID;

@Entity
@Data
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String vehicleId;
    private String address;
    private String startedAt;
    private String finishedAt;
    private String status;
}

File: src/main/kotlin/com/ahold/technl/sandbox/entity/Delivery.kt
package com.ahold.technl.sandbox.entity

import com.ahold.technl.sandbox.dto.DeliveryStatus
import jakarta.persistence.*
import org.jetbrains.annotations.NotNull
import java.time.OffsetDateTime
import java.util.*

@Entity
@Table
data class Delivery(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @field:NotNull
    val vehicleId: String,

    @field:NotNull
    val address: String,

    @field:NotNull
    val startedAt: OffsetDateTime,

    val finishedAt: OffsetDateTime? = null,

    @Enumerated(EnumType.STRING)
    val status: DeliveryStatus
)